### DISCUSSION TITLE
Statefulset vs Deployment

### DISCUSSION THREAD

**Discussion by cdancy** [vectordotdev/helm-charts#333] [General]
Hey All,

Is there a _really good reason_ to use the statefulset option (Aggregator) to deploy vector vs the deployment option (Stateless Aggregator)? I assume you all are persisting _something_, which would make the Aggregator option viable, but what is it exactly your storing and is that necessary, in your opinion, for production deployments?

Thanks,
Chris

**Comment by jszwedko** [comment]
Hi @cdancy ,

Persistence is used for:

- Disk buffers
- Components that checkpoint (`file`, `kubernetes_logs`, and `journald`)

If you aren't using any of those, or don't care if the state is lost, then using the stateless `Aggregator` is preferable to simplify scaling.

**Comment by cdancy** [comment]
@jszwedko our current setup is that we have fluentbit sending logs to vector with vector routing logs to loki/s3/etc/etc. The heavy handling is done on the fluentbit side with vector acting as the "main log router" if you will. In this scenario it feels like we're OK and don't necessarily need a statefulset unless I'm missing something.

**Reply by jszwedko** [reply]
It doesn't _seem_ like you need it. You might find these docs helpful: https://vector.dev/docs/setup/going-to-prod/
